arch/armv8m/tls.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * tls.c
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/types.h>
+
+
+volatile uintptr_t arm_tls_ptr __attribute__((section("armtls"))) = 0;
+
+
+void *read_tls_ptr(void)
+{
+	return (void *)arm_tls_ptr;
+}
+
+
+void __attribute__((naked)) __aeabi_read_tp(void)
+{
+	__asm__ volatile(
```

Comment:
Can we leave it without a space to be compliant with clang-format? Like for other architectures

Reply:
I'd rather not, clang-format is confused in this case, as it thinks it's a function - it's not. It would be best to use clang-format off on every inline asm instead